src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int USER_MIN_AGE = 18;
+    private static final int USER_LOGIN_PASSWORD_MIN_LENGTH = 6;
+    private static final int USER_VALID_AGE = 20;
+    private static final String USER_VALID_LOGIN = "username";
+    private static final String USER_VALID_PASSWORD = "password";
+    private StorageDao storageDao = new StorageDaoImpl();
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User testUser;
+
+    @BeforeEach
+    void beforeEach() {
+        testUser = createValidUser();
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userUnderMinAge_notOk() {
+        testUser.setAge(USER_MIN_AGE - 1);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(testUser));
+        assertEquals(exception.getMessage(), "Users under "
+                + USER_MIN_AGE + " are not allowed!");
+    }
+
+    @Test
+    void register_userWithNullAge_notOk() {
+        User user = createValidUser();
+        user.setAge(null);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Users under "
+                + USER_MIN_AGE + " are not allowed!");
```

Comment:
you can actually see that problem here, but you just covered it with tests that are expecting this strange behavior, we should fix business logic when you face something like this